pointcloud/basic_octree.go

```diff
@@ -140,7 +147,7 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	switch octree.node.nodeType {
 	case internalNode:
 		ocbox, err := spatialmath.NewBox(
-			spatialmath.NewPoseFromPoint(octree.center),
+			spatialmath.NewPoseFromPoint(octree.center.Point()),
```

Comment:
This is not correct.

With `center` being a `Pose`, then `spatialmath.NewPoseFromPoint(octree.center.Point())` should become just `spatialmath.NewPose(octree.center)`

Reply:
could it be just `octree.center` or will that be bad because it's not a copy of the center

Reply:
Ack you're right, it should just be `octree.center`

A `Pose` should always be a pointer to an immutable thing, so reusing the `center` pose is valid to do